client/src/components/NavBar/NotificationsMenu/NotificationsMenu.tsx

```diff
@@ -0,0 +1,113 @@
+import useStyles from './useStyles';
+import { useAuth } from '../../../context/useAuthContext';
+import {
+  Typography,
+  Button,
+  Badge,
+  MenuItem,
+  Menu,
+  Paper,
+  ListItemText,
+  ListItemAvatar,
+  Avatar,
+  ListItem,
+  Box,
+} from '@material-ui/core';
+import { Notification } from '../../../interface/Notification';
+import React, { useEffect, useState } from 'react';
+import { getAllUnreadNotifications, markAsRead } from '../../../helpers/APICalls/Notification';
+
+export default function NotificationsMenu(): JSX.Element {
+  const classes = useStyles();
+  const { loggedInUser } = useAuth();
+
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    notifications?.map(async (notification) => {
+      await markAsRead(notification._id);
+    });
```

Comment:
What I think would be even better, could we just send down a batch of notification ids to the controller and have the server perform batch processing on the notifications and update them through the database driver?